File: SpringBootAOP/src/main/java/com/example/service/TestServiceImpl.java
package com.example.service;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class TestServiceImpl implements TestService {

	private static final Logger logger = LoggerFactory.getLogger(TestServiceImpl.class);

	@Override
	public String test() {
		String msg = "Hello, Spring Boot No AOP";
		logger.info(msg);
		return msg;
	}

	@Override
	public String testAop() {
		String msg = "Hello, Spring Boot AOP";
		logger.info(msg);
		return msg;
	}
}

File: SpringBootAOP/src/main/kotlin/com/example/kotlin/service/TestServiceImpl.kt
package com.example.kotlin.service

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class TestServiceImpl : TestService {

    private val logger = LoggerFactory.getLogger(TestServiceImpl::class.java)

    override fun test(): String {
        val msg = "Hello, Spring Boot With Kotlin No AOP"
        logger.info(msg)
        return msg
    }

    override fun testAop(): String {
        val msg = "Hello, Spring Boot With Kotlin AOP"
        logger.info(msg)
        return msg
    }
}
